src/functions/module_lookup.ts

```diff
@@ -858,3 +866,101 @@ export const PIVOT = {
   },
   returns: ["RANGE<ANY>"],
 } satisfies AddFunctionDescription;
+
+//--------------------------------------------------------------------------
+// OFFSET
+//--------------------------------------------------------------------------
+
+export const OFFSET = {
+  description: _t(
+    "Returns a range reference shifted a specified number of rows and columns from a starting cell reference."
+  ),
+  args: [
+    arg(
+      "cell_reference (meta)",
+      _t("The starting point from which to count the offset rows and columns.")
+    ),
+    arg("offset_rows (number)", _t("The number of rows to offset by.")),
+    arg("offset_columns (number)", _t("The number of columns to offset by.")),
+    arg(
+      "height (number, optional, defaults to the height of cell_reference)",
+      _t("The height of the range to return starting at the offset target.")
+    ),
+    arg(
+      "width (number, optional, defaults to the width of cell_reference)",
+      _t("The width of the range to return starting at the offset target.")
+    ),
+  ],
+  compute: function (
+    cellReference: Maybe<{ value: string }>,
+    offsetRows: Maybe<FPayloadNumber>,
+    offsetColumns: Maybe<FPayloadNumber>,
+    height: Maybe<FPayloadNumber>,
+    width: Maybe<FPayloadNumber>
+  ) {
+    const _cellReference = cellReference?.value;
+    const zone = toZone(_cellReference!);
+    const _offsetRows: number = offsetRows?.value || 0;
+    const _offsetColumns: number = offsetColumns?.value || 0;
+    const _height: number = height ? height.value - 1 : zone.bottom - zone.top;
```

Comment:
you can use an assert to check if height and width are positive or strictly positive ;)